wagtail_localize/views/edit_translation.py

```diff
@@ -414,11 +414,17 @@ def widget_from_block(block, content_components=None):
     if isinstance(field, StreamField):
         block_type_name = field_path_components[1]
         block_type = field.stream_block.child_blocks[block_type_name]
+        content_components = content_path_components[2:]
 
         if isinstance(block_type, blocks.StructBlock):
             block_field_name = field_path_components[2]
             block_field = block_type.child_blocks[block_field_name].label
-
+        elif isinstance(block_type, blocks.StreamBlock):
+            block_field_name = field_path_components[2]
+            block_field = block_type.child_blocks[block_field_name].label
+            # use the field path components instead of content path as we're drilling down
+            # a set of nested blocks
+            content_components = field_path_components[2:]
```

Comment:
this then makes use of the change in models.py

Reply:
A little worried about mixing field/content paths since they will have different values when working with nested streamblocks.

 - Content path is a path within an instance, so stream blocks are represented with their ID
 - Field path is a path on the data model, so stream blocks are represented with their type name (this was introduced to allow us to look up the field a segment was inserted into without having to retrieve the original content from the database)

So if you pass a field path into a function that expects a content path, it'll only work if there aren't any streamblocks within that path. So I think here it could break if there's a nested stream block?

Reply:
Note: tidied up in 1e284bb (during our pair programming session)